test/parallel/test-stream-readable-async-iterators.js

```diff
@@ -0,0 +1,297 @@
+    readable.push('hello-3');
+    readable.push('hello-4');
+    readable.push(null);
+
+    const last = await iter.next();
+    assert.strictEqual(last.done, true);
+  })();
+
+  await (async function() {
+    console.log('read without for..await deferred');
+    const readable = new Readable({
+      objectMode: true,
+      read() {}
+    });
+
+    const iter = readable[Symbol.asyncIterator]();
+    assert.strictEqual(iter.stream, readable);
+    let values = [];
+    for (let i = 0; i < 3; i++) {
+      values.push(iter.next());
+    }
+
+    readable.push('hello-0');
+    readable.push('hello-1');
+    readable.push('hello-2');
+
+    let k = 0;
+    const results1 = await Promise.all(values);
+    results1.forEach(common.mustCall(
+      (item) => assert.strictEqual(item.value, 'hello-' + k++), 3));
+
+    values = [];
+    for (let i = 0; i < 2; i++) {
+      values.push(iter.next());
+    }
+
+    readable.push('hello-3');
+    readable.push('hello-4');
+    readable.push(null);
+
+    const results2 = await Promise.all(values);
+    results2.forEach(common.mustCall(
+      (item) => assert.strictEqual(item.value, 'hello-' + k++), 2));
+
+    const last = await iter.next();
+    assert.strictEqual(last.done, true);
+  })();
+
+  await (async function() {
+    console.log('read without for..await with errors');
+    const max = 3;
+    const readable = new Readable({
+      objectMode: true,
+      read() {}
+    });
+
+    const iter = readable[Symbol.asyncIterator]();
+    assert.strictEqual(iter.stream, readable);
+    const values = [];
+    const errors = [];
+    let i;
+    for (i = 0; i < max; i++) {
+      values.push(iter.next());
+    }
+    for (i = 0; i < 2; i++) {
+      errors.push(iter.next());
+    }
+
+    readable.push('hello-0');
+    readable.push('hello-1');
+    readable.push('hello-2');
+
+    const resolved = await Promise.all(values);
+
+    resolved.forEach(common.mustCall(
+      (item, i) => assert.strictEqual(item.value, 'hello-' + i), max));
+
+    errors.forEach((promise) => {
+      promise.catch(common.mustCall((err) => {
+        assert.strictEqual(err.message, 'kaboom');
+      }));
+    });
+
+    readable.destroy(new Error('kaboom'));
+  })();
+
+  await (async function() {
+    console.log('read object mode');
+    const max = 42;
+    let readed = 0;
+    let received = 0;
+    const readable = new Readable({
+      objectMode: true,
+      read() {
+        this.push('hello');
+        if (++readed === max) {
+          this.push(null);
+        }
+      }
+    });
+
+    for await (const k of readable) {
+      received++;
+      assert.strictEqual(k, 'hello');
+    }
+
+    assert.strictEqual(readed, received);
+  })();
+
+  await (async function() {
+    console.log('destroy sync');
+    const readable = new Readable({
+      objectMode: true,
+      read() {
+        this.destroy(new Error('kaboom from read'));
+      }
+    });
+
+    let err;
+    try {
+      /*eslint no-unused-vars: 0*/
```

Comment:
i think this disables the rule for the rest of the file. Instead you can use `// eslint-disable-next-line no-unused-vars`